stl/inc/xstring

```diff
@@ -4600,6 +4616,31 @@ template <class _Elem, class _Traits, class _Alloc>
 _NODISCARD bool operator>=(const basic_string<_Elem, _Traits, _Alloc>& _Left, _In_z_ const _Elem* const _Right) {
     return !(_Left < _Right);
 }
+#endif // !_HAS_CXX20
+
+#if _HAS_CXX20
+template <class _Elem, class _Traits, class _Alloc>
+_NODISCARD strong_ordering operator<=>(
+    const basic_string<_Elem, _Traits, _Alloc>& _Left, const basic_string<_Elem, _Traits, _Alloc>& _Right) noexcept {
+    int _comparison = _Left.compare(_Right);
+    if (_comparison == 0) {
+        return strong_ordering::equivalent;
+    } else if (_comparison < 0) {
+        return strong_ordering::less;
+    }
+    return strong_ordering::greater;

```

Comment:
This could be `return static_cast<R>(_Left.compare(_Right) <=> 0)` as specified in the standard, where _R_ is the return type.

Same below.